rllib/agents/a3c/a2c_workflow.py

```diff
@@ -0,0 +1,33 @@
+"""Experimental workflow-based impl; run this with --run='A2C_wf'"""
+
+import math
+
+from ray.rllib.agents.a3c.a2c import A2CTrainer
+from ray.rllib.utils.experimental_dsl import (
+    ParallelRollouts, ConcatBatches, ComputeGradients, AverageGradients,
+    ApplyGradients, TrainOneStep, StandardMetricsReporting)
+
+
+def training_workflow(workers, config):
+    rollouts = ParallelRollouts(workers, mode="bulk_sync")
+
+    if config["microbatch_size"]:
+        num_microbatches = math.ceil(
+            config["train_batch_size"] / config["microbatch_size"])
+
+        train_op = rollouts \
+            .combine(ConcatBatches(min_batch_size=config["microbatch_size"])) \
+            .for_each(ComputeGradients(workers)) \
+            .batch(num_microbatches) \
+            .for_each(AverageGradients()) \
+            .for_each(ApplyGradients(workers))
```

Comment:
it's hard for me to understand what "each" refers to - if it is referring to batch, what about "for batch"? "for each batch"?

Reply:
How about .then()?

This is coming from the parallel iterators API, so it is not specific to RL terms (i.e., should not call it for_batch).